src/crush/CrushWrapper.cc

```diff
@@ -982,7 +982,7 @@ int CrushWrapper::verify_upmap(CephContext *cct,
           return -EINVAL;
         }
         // validate the osd's in subtree
-        for (int c = 0; cursor < up.size() && c < numrep; ++cursor, ++c) {
+        for (int c = 0; static_cast<uint64_t>(cursor) < up.size() && c < numrep; ++cursor, ++c) {
```

Comment:
just 2 nits:

- `vector::size()` returns `std::size_t`, and on 32bit platforms, `size_t` is always 32bits. so, probably we can just change the type of `cursor` to `size_t` or `std::size_t` ?
- please wrap at 80 chars. but if the first suggestion makes sense to you, this one is not necessary anymore.

Reply:
Thanks. size_t makes sense here. Updated PR.